ref-arch/real-time-analytics.rst

```diff
@@ -0,0 +1,434 @@
+the raw data around forever is prohibitively expensive.
+
+Or maybe you're building an advertising network and want to show clients clickthrough
+rates on their campaigns. Likewise, you want to ingest lots of data with as little latency
+as possible and show both historical and live data on a dashboard.
+
+In this reference architecture we'll demonstrate how to build part of the first example
+but this architecture would work equally well for the second and many other business
+use-cases.
+
+Running It Yourself
+-------------------
+
+There's `a github repo <http://github.com>`_ with scripts and usage instructions. If
+you've gone through our installation instructions for running on either single or multiple
+machines you're ready to try it out. There will be some code snippets in this tutorial
+but the github repo has all the details in one place.
+
+Data Model
+----------
+
+The data we're dealing with is an immutable stream of log data. Here we'll insert directly
+into Citus but it's also common for this data to first be routed through something like
+Kafka. Doing so makes the system a little more resilient to failures, lets the data be
+routed to multiple places (such as a warehouse like redshift), and (once data volumes
+become unmanageably high) makes it a little easier to start pre-aggregating the data
+before inserting.
+
+In this example, the raw data will use the following schema which isn't very realistic as
+far as http analytics go but sufficient for showing off the architecture we have in mind.
+
+.. code-block:: sql
+
+  -- this gets run on the master
+  CREATE TABLE http_request (
+    zone_id INT,
+    ingest_time TIMESTAMPTZ DEFAULT now(),
+
+    session_id UUID,
+    url TEXT,
+    request_country TEXT,
+    ip_address CIDR,
+
+    status_code INT,
+    response_time_msec INT,
+  )
+  SELECT master_create_distributed_table('http_request', 'zone_id', 'hash');
+  SELECT master_create_worker_shards('http_request', 16, 2);
+
+When we call :ref:`master_create_distributed_table <master_create_distributed_table>`
+we're telling it to hash-distribute using the `zone_id` column. That means dashboard
+queries will each hit a single shard.
+
+When we call :ref:`master_create_worker_shards <master_create_worker_shards>` we tell it
+to create 16 shards, and 2 copies of each shard. :ref:`We recommend
+<faq_choose_shard_count>` using 2-4x as many shards as cores in your cluster, this reduces
+the overhead of keeping track of lots of shards but also leaves you plenty of shards to
+spread around when you scale up your cluster by adding workers.
+
+Using a replication factor of 2 (or any number > 1, really), means data is written to
+multiple workers. When a node fails the worker will serve queries for a shard using the
+other node so you don't have any downtime.
+
+.. NOTE::
+
+  In Citus Cloud you must use a replication factor of 1 (instead of the 2 used here), as
+  they have a different HA solution.
```

Comment:
As mentioned in the call, I'd word this as:

```
In Citus Cloud you must use a replication factor of 1 (instead of the 2 used here), as Citus Cloud uses PostgreSQL streaming replication for high availability.
```